File: front-chaveamento-robos/src/types/torneio.ts
import type { EnumTime, Time } from "./time";

export type EnumTorneio = EnumTime;

export type StatusTorneio = "INSCRICOES" | "EM_ANDAMENTO" | "FINALIZADO";

export interface Torneio {
  id: number;
  tipo: EnumTorneio;
  status: StatusTorneio | null;
}

export interface CriarTorneioRequest {
  tipo: EnumTorneio;
}

export interface ParticipacaoTorneio {
  id: number;
  torneio: Torneio;
  time: Time;
}

File: chaveamento-robos/src/main/java/org/chaveamento/model/torneio/Torneio.java
package org.chaveamento.model.torneio;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "torneios")
public class Torneio {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    private EnumTorneio tipo;

    @Enumerated(EnumType.STRING)
    private StatusTorneio status;

    public Torneio(EnumTorneio tipo) {
        this.tipo = tipo;
        this.status = StatusTorneio.INSCRICOES;
    }
}
